puppet/lib/puppet/reports/puppetdb.rb

```diff
@@ -73,6 +75,41 @@ def build_events_list
     end
   end
 
+  # @return Array[Hash]
+  # @api private
+  def build_logs_list
+    profile("Build logs list (count: #{logs.count})",
+            [:puppetdb, :logs_list, :build]) do
+      log_list = []
+      logs.each do |log|
```

Comment:
```logs.map do |log|``` ... should remove the need for both vars here: log_list for obvious reasons, and log_hash because the last value in a block is what gets returned.